# Import Library

In [1]:
import pandas as pd
import pickle

# Load Model

In [2]:
with open('best_model.pkl', 'rb') as file:
    best_model = pickle.load(file)

# Model Inference

In [11]:
data = {
    'name': 'Honda CRV',
    'year': 2018,
    'km_driven' : 10000,
    'fuel' : 'Petrol',
    'seller_type' : 'Individual',
    'transmission' : 'Automatic',
    'owner' : 'First Owner',
    'mileage' : 15,
    'engine': 1200,
    'max_power' : 100,
    'seats' : 5,
    'brand' : 'Honda'
}

data = pd.DataFrame([data])

In [12]:
target_pred = best_model.predict(data)

print(f'Prediksi Harga Mobil bekas Honda CRV tahun 2018 adalah {target_pred} Rupee')

Prediksi Harga Mobil bekas Honda CRV tahun 2018 adalah [771396.34082187] Rupee


# Conclusion

This model was built to help users who want to sell their used car get an objective price estimate. From the data we had, we could see that there were some extreme price values in the used car dataset. After all the data preprocessing and model tuning, the model shows that for a Honda CRV year 2018, the predicted used car price is approximately 771,396 Rupee.

This means the model can be used as a practical tool to give sellers a data-driven price reference, rather than relying on guesswork or personal opinion.